redaxo/src/addons/structure/lib/navigation.php

```diff
@@ -310,6 +327,10 @@ private function checkFilter(rex_category $category, $depth)
     }
 
     /**
+     * @param int $depth
+     * @param array<int|string, int|string|list<string>> $li
+     * @param array<int|string, int|string|list<string>> $a
+     * @param string $aContent
      * @return bool
      */
     private function checkCallbacks(rex_category $category, $depth, &$li, &$a, &$aContent)
```

Comment:
by-ref paramer könntest du zusätzlich mit `@param-out` beschreiben

Reply:
Hatte ich auch überlegt, aber hatte dann in der psalm doku das gelesen:

> This is used to specify that a by-ref type is different from the one that entered.

https://psalm.dev/docs/annotating_code/supported_annotations/#param-out-psalm-param-out

Und hatte es dann weglassen, da sich der rausgehende Typ nicht unterscheidet.

Reply:
Zumindest in phpstan ist ein by ref param immer mixed auf der call-site

Reply:
Oh ok, dann macht es natürlich tatsächlich Sinn die zu notieren